### ISSUE TITLE
fix(rule): DailyLossLimitRule 손실률 분모를 전일 총 자산으로 수정

### ISSUE THREAD

**Issue by joshua-jingu-lee** [joshua-jingu-lee/ante#708]
## 배경

`DailyLossLimitRule`의 일일 손실률 산식이 스펙(`docs/specs/rule-engine.md` 176행)과 불일치한다.

- **스펙**: `일일 손실률 = 당일 실현 손익 / 전일 총 자산 × 100`
- **구현**: `daily_loss_amount / (total_pnl + daily_loss_amount)` — 분모가 "총 손익"

### 결함

1. **비율 왜곡** — 자산 1억, 총 손익 100만 상태에서 50만 원 손실 시: 스펙 0.5% vs 구현 33.3%
2. **안전장치 무력화** — `total_pnl == 0`이면 검사를 건너뜀 (14행)
3. **음수 분모** — 총 손익이 음수면 `base > 0` 조건에서 차단되어 룰 무시

ref: `docs/temp/spec-gap-002-daily-loss-limit-formula.md`

### 1. `RuleContext`에 `prev_day_total_asset` 필드 추가

**파일**: `src/ante/rule/base.py`

```python
# 계좌 상태 (기존)
account_status: str = "active"
daily_pnl: float = 0.0
total_pnl: float = 0.0
prev_day_total_asset: float = 0.0  # ← 추가
```

### 2. `DailyLossLimitRule.evaluate()` 산식 수정

**파일**: `src/ante/rule/global_rules.py`

```python
def evaluate(self, context: RuleContext) -> RuleEvaluation:
    max_daily_loss = self.config.get("max_daily_loss_percent", 0.05)

    if context.daily_pnl < 0 and context.prev_day_total_asset > 0:
        daily_loss_percent = abs(context.daily_pnl) / context.prev_day_total_asset

        if daily_loss_percent > max_daily_loss:
            return RuleEvaluation(
                rule_id=self.rule_id,
                rule_name=self.name,
                result=RuleResult.BLOCK,
                action=RuleAction.HALT_ACCOUNT,
                message=(
                    f"Daily loss limit exceeded: "
                    f"{daily_loss_percent:.2%} > {max_daily_loss:.2%}"
                ),
                metadata={
                    "daily_loss_percent": daily_loss_percent,
                    "max_daily_loss_percent": max_daily_loss,
                    "daily_pnl": context.daily_pnl,
                    "prev_day_total_asset": context.prev_day_total_asset,
                },
            )

    return RuleEvaluation(
        rule_id=self.rule_id,
        rule_name=self.name,
        result=RuleResult.PASS,
        action=RuleAction.LOG,
        message="Daily loss within limits",
    )
```

**변경 요점**:
- 분모: `total_pnl + daily_loss_amount` → `prev_day_total_asset`
- `total_pnl != 0` 조건 → `prev_day_total_asset > 0` 조건 (전일 자산이 0이면 의미 없으므로)
- metadata에 `prev_day_total_asset` 추가

### 3. `RuleEngine._build_context()`에서 전일 총 자산 주입

**파일**: `src/ante/rule/engine.py` (296행, 441행의 `RuleContext` 생성부)

`RuleEngine`이 `RuleContext`를 생성할 때 `prev_day_total_asset`을 주입해야 한다.
데이터 소스는 Treasury 또는 Account 모듈의 전일 정산 스냅샷이다.

### 4. 테스트 수정

**파일**: `tests/unit/test_rule.py`

- `TestDailyLossLimitRule` 기존 테스트: `total_pnl` 대신 `prev_day_total_asset` 사용하도록 수정
- 추가 케이스:
  - `prev_day_total_asset > 0`일 때 정상 차단/통과
  - `prev_day_total_asset == 0`일 때 룰 PASS (검사 불가)
  - 대규모 자산 대비 소규모 손실 → PASS (기존 구현에서는 오탐 발생하던 케이스)

## 수정 대상 파일

- `src/ante/rule/base.py` — `RuleContext.prev_day_total_asset` 필드 추가
- `src/ante/rule/global_rules.py` — `DailyLossLimitRule.evaluate()` 산식 수정
- `src/ante/rule/engine.py` — `RuleContext` 생성 시 `prev_day_total_asset` 주입
- `tests/unit/test_rule.py` — 테스트 수정 및 추가

**Comment by joshua-jingu-lee** [comment]
🤖 **구현 착수**
- 담당 에이전트: @backend-dev
- 변경 대상: src/ante/rule/base.py, src/ante/rule/global_rules.py, src/ante/rule/engine.py, tests/unit/test_rule.py
- base 브랜치: main

**Comment by joshua-jingu-lee** [comment]
🤖 **코드 리뷰: APPROVE**
- PR: #709
- 체크리스트 A~F 전체 통과

**Comment by joshua-jingu-lee** [comment]
## 구현 갭 발견: RuleEngine ↔ Treasury 연동 미비

### 문제

`RuleContext`에 `prev_day_total_asset` 필드를 추가하고 산식을 수정하더라도, `RuleEngine._on_order_request()`에서 해당 값을 실제로 주입하지 않으면 항상 `0.0`이므로 `prev_day_total_asset > 0` 조건을 통과하지 못해 **룰이 항상 PASS**됩니다.

### 현재 상태

- `RuleEngine`은 `RuleContext` 생성 시 `AccountService`만 의존성으로 사용 중
- `daily_pnl`, `total_pnl` 역시 현재 항상 `0.0` — 기존 룰도 사실상 미작동 상태
- 데이터 소스(`Treasury.get_daily_snapshot()`, `Treasury.get_summary()`)는 이미 존재하나, RuleEngine과의 연동이 없음

### 필요 작업

본 이슈(§3)의 "RuleEngine에서 전일 총 자산 주입"은 단순 필드 매핑이 아니라, **RuleEngine이 Treasury를 의존성으로 받아 스냅샷을 조회하는 연동 작업**이 선행되어야 함.

이 연동은 #710(`TotalExposureLimitRule`의 `total_asset`, `total_exposure`)에도 공통으로 필요한 선행 작업이므로, PR #709는 리젝하고 연동 작업을 포함한 재구현이 필요합니다.

**Comment by joshua-jingu-lee** [comment]
선행 이슈 등록: #711 (RuleEngine ↔ Treasury 연동) 완료 후 본 이슈 재구현 가능.

**Comment by joshua-jingu-lee** [comment]
🤖 **구현 재착수**
- 담당 에이전트: @backend-dev
- 변경 대상: src/ante/rule/base.py, src/ante/rule/global_rules.py, src/ante/rule/engine.py, src/ante/rule/manager.py, src/ante/main.py, tests/unit/test_rule.py
- 추가 범위: RuleEngine ↔ Treasury 연동 (prev_day_total_asset + daily_pnl 실제 주입)
- base 브랜치: main

**Comment by joshua-jingu-lee** [comment]
🤖 **코드 리뷰: APPROVE**
- PR: #712
- 체크리스트 A~F 전체 통과
- Nit: engine.py 내 `from datetime` lazy import → 파일 상단 이동 권장 (비차단)
